### DISCUSSION TITLE
Development ENV

### DISCUSSION THREAD

**Discussion by massimoliani** [directus/directus#9216] [Archive]
Hi,

I'm developing an app with directus, but I need to develop extensions for the frontend and I would like to use the development version to debug, for the first displays and interfaces I use only console.log but is tedious and long to do.

I configure the development env, I make an hard link to src extensions folder of original project and source .env files.
It work correctly but the extensions to be compiled aren't loaded, like displays and interfaces for example.

How can I link the source folder and have the watch processes compile for debug the extensions? 
Is there a preferred location or env variable to use?

Thank you.

**Comment by joselcvarela** [comment]
Hello @massimoliani 
When I need to develop a custom display or interface I use this command `npx directus-extension build -w` and modify the destination in `package.json`. This allows to run build on every change and, since it has the right path in `package.json`, I just need to refresh the browser.

I don't know if it is possible to run Directus locally and develop custom interfaces without bundling it first, but I think @nickrum could help you better.

**Reply by massimoliani** [reply]
Thank @joselcvarela could be possible to build with debug information inside to use Vue browser extension for debugging?

**Reply by nickrum** [reply]
This is not currently possible. The extension source has to bundled for Directus to be able to load it. We will provide a dedicated extension dev environment in the future, though.

**Reply by massimoliani** [reply]
Is it possible to enable the vue devtools of chrome? Or the debug data are normally stripped from the extension?

Thank you.

**Reply by cpt-n3mo** [reply]
Good day!
massimoliani, its indeed a little cumbersome to read minified JS, i found that firefox devtools not working very wel..  but chrome devtools works abit better in my opinion, other thing you can do to not have to restart directus after every change is that you can use nodemon to watch the direcuts files.. `nodemon --exec npx directus start ` you can ignore files or folders by adding a `--ignore folder_or_file` if you want multiple files or folders to be ignored, then use multiple `--ignore`

togeather with what joselcvarela was saying is what i use to dev a module,.. and works quite well.. you only need to refresh the browser this way to see the changes.. it will still give you minified JS, but with a little puzzle work its usually findable where a problem is 
kind regards!

**Reply by massimoliani** [reply]
Thank you @cpt-n3mo I found in directus dev env you use npm-watch to hot reload, I'll test nodemon too.
Could be great to have in the future the possibility to avoid JS minifying :-) 

As I start in '80 to write software I'm used to debug using log messages too :-) 

Directus it's a great architecture to develop generic web app too having a running backend with admin dashboard.

**Reply by cpt-n3mo** [reply]
ah thats prob new then... perfect this looks like a vite server so its prob hotreload then as well.. no browser refresh...? that would be awesome! and im pritty sure they will find a way to give us non minified JS, the project is growing by the day it seems :)

and indeed this architecture/framework is just awesome, im developing a module to allow for a live nuxt dev server inside directus. it was allot of proxy work but it works like a charm, point click and edit function, with static generator function, i always wanted to make such a system, and directus made that possible taking all the heavy lifting away  :)  so a very happy user here :)
chears!!

**Reply by barakplasma** [reply]
the `directus-extension build` command seems to be built on top of Rollup as of this comment, so I modified my package.json scripts like so
`"build": "directus-extension build --no-minify --sourcemap",` to avoid the minification step and to add a sourcemap.
